src/lib/common/sem.h

```diff
@@ -65,6 +65,23 @@ extern int semInit
 
 /* ****************************************************************************
 *
+* reqSemTryToTake - try to take take semaphore
+*
+* This function is only used by the exit-function, in order to clean the subscription cache.
+* The exit-function runs when the broker is shutting down and instead of just wating for the
+* semaphore, it is taken if it is free, if not, the subscription cache is simply cleaned.
+* A little dangerous, no doubt, but, the broke is shutting down anyway, if it dies because
+* of a SIGSEGV in the subscription cache code, it is not such a big deal ... perhaps ...
+*
+* FIXME P4
+* This implementation will stay like this until we find a better way attack the problem.
```

Comment:
My concern is that SIGSEGV errors appear in production (e.g. when the service is restarted) controled by a team that doesn't know this deep detail.

What about not cleaning the cache in production at exitFunction and use this only in DEBUG (for valgrind) using conditional compilation?